ethpm/utils/package_validation.py

```diff
@@ -22,20 +27,42 @@ def _load_schema_data():
         return json.load(schema)
 
 
-def load_and_validate_package(package_id):
+def validate_package_against_schema(package_data):
     """
     Load and validate package json against schema
     located at RELEASE_LOCKFILE_SCHEMA_PATH.
     """
     schema_data = _load_schema_data()
-    package_data = _load_package_data(package_id)
     try:
         validate(package_data, schema_data)
     except jsonValidationError:
         raise ValidationError(
-            "Package:{0} invalid for schema:{1}".format(package_id, RELEASE_LOCKFILE_SCHEMA_PATH)
+            "Package:{0} invalid for schema:{1}".format(package_data, RELEASE_LOCKFILE_SCHEMA_PATH)
         )
-    return package_data
+
+
+def validate_package_deployments(package_data):
+    """
+    Validate that a package's deployments contracts reference existing contract_types.
+    """
+    if set(("contract_types", "deployments")).issubset(package_data):
+        all_contract_types = list(package_data["contract_types"].keys())
+
+        deployments = [
+            deployment
+            for uri, deployment
+            in package_data["deployments"].items()
+        ]
+        deployment_names = [
+            name
+            for name, value
+            in deployments[0].items()
```

Comment:
This will only check the 1st set of deployments.  I think you want to change to something like

```
all_deployment_names = set(itertools.chain.from_iterable(
    deployments.keys()
    for deployments
    in all_deployments
))
```